Homework_2,py

```diff
@@ -0,0 +1,42 @@
+# Homework_02
+
+# 1. create a list of random number of dicts (from 2 to 10)
+# dict's random numbers of keys should be letter,
+# dict's values should be a number (0-100),
+# example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+
+import random #import library for useing random function
+import string #import library for working with strings
+
+# generate a list of dictionaries
+# n - number of dictionaries (list elements)
+# m - number of elements in each dictionary
+res_list = [({random.choice(string.ascii_lowercase): random.randint(0, 100) for m in range(random.randint (1, 26))}) for n in range(random.randint(2, 10))]
+
+print(res_list)
+
+# 2. get previously generated list of dicts and create one common dict:
+# if dicts have same key, we will take max value, and rename key with dict number with max value
+# if key is only in one dict - take it as is,
+# example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+
+# checking res_list=[{'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}]
+
+final_dict = {} # final dictionary for adding the result
+
+for char in string.ascii_lowercase: # go through all letters in the alphabet
+    temp_values={} # temporary dictionary for each letter with the number of dictionary and value for this key
+    for i in enumerate(res_list): # circle through all dictionaries in the list
```

Comment:
you can use enumerate in a more efficient way like `for ind, next_dict in enumerate(res_list):` so you can easily access index and object of iteration inside the cycle

Reply:
Done